ui/src/app/views/copy-notebook-modal/component.tsx

```diff
@@ -0,0 +1,185 @@
+          loading: false
+        });
+      });
+  }
+
+  save() {
+    this.setState({ loading: true });
+
+    workspacesApi().copyNotebook(
+      this.props.fromWorkspaceNamespace,
+      this.props.fromWorkspaceName,
+      this.props.fromNotebook.name,
+      {
+        toWorkspaceName: this.state.destination.id,
+        toWorkspaceNamespace: this.state.destination.namespace,
+        newName: this.state.newName
+      }
+      ).then((response) => {
+        this.setState({ state: State.SUCCESS, loading: false });
+        this.props.onCopy(response);
+      }).catch((response) => {
+        const errorMsg = response.status === 400 ?
+          'Notebook with the same name already exists in the targeted workspace.' :
+          'An error occurred while copying. Please try again.';
+
+        this.setState({
+          errorMsg: errorMsg,
+          state: State.ERROR,
+          loading: false
+        });
+      });
+  }
+
+  goToDestinationWorkspace() {
+    window.location.href = this.getWorkspaceUrl(this.state.destination);
+  }
+
+  getWorkspaceUrl(workspace: Workspace) {
+    return ['workspaces', workspace.namespace, workspace.id, 'notebooks'].join('/');
+  }
+
+  render() {
+    return (
+      <Modal onRequestClose={this.props.onClose}>
+        <ModalTitle>Copy to Workspace</ModalTitle>
+          { this.state.loading ?
+            <ModalBody style={{ textAlign: 'center' }}><Spinner /></ModalBody> :
+            <ModalBody>
+                {this.state.state === State.FORM && this.renderFormBody()}
+                {this.state.state === State.ERROR && this.renderErrorBody()}
+                {this.state.state === State.SUCCESS && this.renderSuccessBody()}
+            </ModalBody>
+        }
+        <ModalFooter>
+          <Button type='secondary' onClick={this.props.onClose}>
+              {this.getCloseButtonText()}
+          </Button>
+          {this.renderActionButton()}
+        </ModalFooter>
+      </Modal>
+    );
+  }
+
+  getCloseButtonText() {
+    if (this.state.state === State.FORM) {
+      return 'Cancel';
+    } else if (this.state.state === State.ERROR) {
+      return 'Close';
+    } else if (this.state.state === State.SUCCESS) {
+      return 'Stay Here';
+    }
+  }
+
+  renderActionButton() {
+    if (this.state.state === State.FORM) {
+      return (
+        <Button style={{ marginLeft: '0.5rem' }}
+          disabled={this.state.destination === null}
+          onClick={() => this.save()}>
+          Copy Notebook
+        </Button>
+      );
+    } else if (this.state.state === State.ERROR) {
+      return null;
+    } else if (this.state.state === State.SUCCESS) {
+      return (
+        <Button style={{ marginLeft: '0.5rem' }}
+          onClick={this.goToDestinationWorkspace}>
+          Go to Copied Notebook
+        </Button>
+      );
+    }
+  }
+
+  renderFormBody() {
+    return (
+      <div>
+        <div style={headerStyles.formLabel}>Destination *</div>
```

Comment:
is the `*` to indicate a required field?  i believe that we have a way to use that using `validate.js` that will prevent the user from submitting if required fields aren't filled out.  see `rename-modal` for example

Reply:
I took a look at `validate.js` and it seems like a bit of overhead for what I need which is just to check if the user selected anything. I'm not doing any smart validation on what the string needs to look like and I believe thats the value add of using `validate.js`